Update http client version

tvernum opened the pull request:
Moves a few Apache HTTP client dependencies to their latest version

- httpclient -> 4.5.13
- httpasyncclient -> 4.1.5
- httpcore -> 4.4.13

tvernum committed:
Update http client version

Moves a few Apache HTTP client dependencies to their latest version

- httpclient -> 4.5.13
- httpasyncclient -> 4.1.5
- httpcore -> 4.4.13

tvernum commented:
@dakrone I'm looking at upgrading httpclient / httpcore to the latest available version

The move to httpcore 4.4.13 has changed the `PublicSuffixMatcher` which means that `RegisteredDomainProcessorTests` fails because `googleapis.com` now has a registered domain (`googleapis.com`) but in the older version it did not. 
- https://gradle-enterprise.elastic.co/s/dxh3m5fny5trw 

Is that something that you would be concerned about? Can we safely just change the test to reflect the new behaviour or do we need to do something more?

dakrone commented:
> Is that something that you would be concerned about? Can we safely just change the test to reflect the new behaviour or do we need to do something more?

I looked at this, I think we can change the test to reflect the new domain list. I don't think this will affect anything otherwise.

tvernum committed:
Merge branch 'master' into httpclient-4-5-13

tvernum committed:
Reflect private suffix matching rules in tests

Since https://github.com/apache/httpcomponents-client/pull/203
httpclient handles private suffix rules differently.

This change updates RegisteredDomainProcessorTests to reflect those
new rules

tvernum commented:
@dakrone It looks like the change is going to be reasonably extensive, and will undo most of the tests that were added in https://github.com/elastic/elasticsearch/pull/82898

Since https://github.com/apache/httpcomponents-client/pull/203 `PRIVATE` domain suffixes are handled differently, 

I've pushed 132680ca8ee that makes the test pass. Can you confirm that you're happy to have that scope of change in a minor?

tvernum committed:
Spotless

dakrone commented:
That seems reasonable to me, but maybe @justincr-elastic can take a look since he did the original work on the test?

tvernum commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'master' into httpclient-4-5-13

elasticmachine commented:
Pinging @elastic/es-core-infra (Team:Core/Infra)

elasticmachine commented:
Pinging @elastic/es-data-management (Team:Data Management)

tvernum commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'master' into httpclient-4-5-13

justincr-elastic commented:
The tests were added to confirm googleapis.com was an eTLD, and not the same as a DNS address. If the updated HTTP Client version includes updates of its bundled list of eTLDs, then adjusting the tests makes sense. It may have side effects on https://www.elastic.co/guide/en/elasticsearch/reference/current/registered-domain-processor.html, but that is expected.

justincr-elastic commented:
There was no bug fix related to adding those tests. The tests were added to confirm how the library is parsing googleapis.com as an eTLD. If the library is paring googleapis.com differently, it is OK to update the tests to reflect that change and make them pass. The tests confirm how it works, and can be used as a reference when questions come up for how the HTTP Client library is parsing eTLDs like googleapis.com.

justincr-elastic reviewed (approved):
LGTM

dakrone reviewed (approved):
LGTM also

tvernum commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'master' into httpclient-4-5-13

tvernum commented:
Grrr, clicked the wrong button :(

tvernum committed:
Update docs/changelog/87491.yaml

elasticsearchmachine commented:
Hi @tvernum, I've created a changelog YAML for you.

tvernum committed:
Update changelog

tvernum committed:
Merge branch 'master' into httpclient-4-5-13
